react/ProductSummary.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Button from '@vtex/styleguide/lib/Button'
+import { FormattedMessage } from 'react-intl'
+
+import ProductName from './ProductName'
+import Price from './Price'
+import DiscountBadge from './DiscountBadge'
+
+/**
+ * Product Summary component. Summarizes the product informations.
+ */
+class ProductSummary extends Component {
+  /**
+   * Constructs the state of the component. Uses a hovering property to react if the Buy button should be shown only on hover.
```

Comment:
⛏ particularmente, acho os comentarios nas funções privadas desnecessarios.

Reply:
Essa explicação já existe na definição das `props` Concordo com @klzns  Pensa sempre na complexidade da função, se houver alguma complexidade fora do padrão, ai faz sentido ter comentário. Mesmo quando for privado.

Reply:
Na verdade, este comentário está explicando a propriedade utilizada no state para não confundir com a propriedade `showButtonOnHover`. Mas tudo bem. Retirarei o comentário.